node/actors/network/src/metrics.rs

```diff
@@ -245,58 +253,58 @@ impl MeteredStreamStats {
             received: 0.into(),
             established: SystemTime::now(),
             peer_addr,
-            sent_buffer: VecDeque::new(),
-            received_buffer: VecDeque::new(),
+            current_minute_sent: 0.into(),
+            previous_minute_sent: 0.into(),
+            current_minute_received: 0.into(),
+            previous_minute_received: 0.into(),
+            current_minute_start: 0.into(),
         }
     }
 
-    fn read(&mut self, amount: u64) {
+    fn read(&self, amount: u64) {
+        self.update_minute();
         self.received.fetch_add(amount, Ordering::Relaxed);
-        self.received_buffer.push_back((Instant::now(), amount));
-        self.cleanup_buffers();
+        self.current_minute_received
+            .fetch_add(amount, Ordering::Relaxed);
     }
 
-    fn wrote(&mut self, amount: u64) {
+    fn wrote(&self, amount: u64) {
+        self.update_minute();
         self.sent.fetch_add(amount, Ordering::Relaxed);
-        self.sent_buffer.push_back((Instant::now(), amount));
-        self.cleanup_buffers();
+        self.current_minute_sent
+            .fetch_add(amount, Ordering::Relaxed);
     }
 
-    fn cleanup_buffers(&mut self) {
-        let one_minute_ago = Instant::now() - Duration::from_secs(60);
+    fn update_minute(&self) {
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards")
+            .as_secs();
 
-        while let Some(&(timestamp, _)) = self.sent_buffer.front() {
-            if timestamp < one_minute_ago {
-                self.sent_buffer.pop_front();
-            } else {
-                break;
-            }
-        }
+        let elapsed_secs = now.saturating_sub(self.current_minute_start.load(Ordering::Relaxed));
```

Comment:
if elapsed_secs is negative, then it means that the system time has been reset, in which case you should reset all the counters.

Reply:
note that alternatively you can use the monotone clock (i.e. Instant as in previous version) and measure time relative to the time when the socket has been opened (i.e. call time::now() in constructor, and then use this timestamp instead of UNIX_EPOCH).